### PR TITLE
HIP-584: Keep track of newly created accounts from HederaEvmStackedWorldStateUpdater in StackedStateFrames

### LINKED ISSUE #6019: HIP-584: Keep track of newly created accounts from `HederaEvmStackedWorldStateUpdater` in `StackedStateFrames`
### Problem

Currently we create new accounts in `HederaEvmStackedWorldStateUpdater.createAccount()`, however we don't insert these accounts in the in-memory state and they might be lost in following nested transactions.

### Solution

Insert newly created accounts in state.

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by 0xivanov**
**Description**:

Currently we create new accounts in HederaEvmStackedWorldStateUpdater.createAccount(), however we don't insert these accounts in the in-memory state and they might be lost in following nested transactions.

**Related issue(s)**:

Fixes #6019

**Notes for reviewer**:
Contains not merged changes from #6066

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/store/contract/HederaEvmStackedWorldStateUpdater.java [hunk 1]
```diff
@@ -78,4 +87,32 @@ public EvmAccount getAccount(final Address address) {
     private boolean isTokenRedirect(final Address address) {
         return hederaEvmEntityAccess.isTokenAccount(address) && evmProperties.isRedirectTokenCallsEnabled();
     }
+
+    private void persistInStackedStateFrames(Address address, Wei balance) {
+        // create new RWCachingStateFrame
+        stackedStateFrames.push();
```
Comment (IvanKavaldzhiev) [inline hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/store/contract/HederaEvmStackedWorldStateUpdater.java:93]: This class should have been already created with a stackedStateFrames that has a new RWCachingStateFrame on top. There is a separate [task](https://github.com/hashgraph/hedera-mirror-node/issues/6029) for implementing this behaviour.

TLDR - Use the stackedStateFrames directly, without push()

File: hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/store/contract/HederaEvmStackedWorldStateUpdater.java [hunk 2]
```diff
@@ -78,4 +87,32 @@ public EvmAccount getAccount(final Address address) {
...
+                0,
+                Collections.emptySortedMap(),
+                Collections.emptySortedMap(),
+                Collections.emptySortedSet(),
+                0,
+                0,
+                0);
+        accountAccessor.set(address, accountModel);
+        // if the current frame's upstream is NOT ROCachingStateFrame - commit and pop
+        if (stackedStateFrames.height() > 1) {
```
Comment (IvanKavaldzhiev) [inline hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/store/contract/HederaEvmStackedWorldStateUpdater.java:113]: We would `commit()` in a single method as it's done in services. The same [task](https://github.com/hashgraph/hedera-mirror-node/issues/6029) would cover this change.

File: hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/store/contract/HederaEvmStackedWorldStateUpdater.java [hunk 3]
```diff
@@ -78,4 +87,32 @@ public EvmAccount getAccount(final Address address) {
...
+                0,
+                0,
+                null,
+                0,
+                Collections.emptySortedMap(),
+                Collections.emptySortedMap(),
+                Collections.emptySortedSet(),
+                0,
+                0,
+                0);
```
Comment (IvanKavaldzhiev) [inline hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/store/contract/HederaEvmStackedWorldStateUpdater.java:110]: We should set also the nonce that is passed.

File: hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/store/contract/HederaEvmStackedWorldStateUpdater.java (hunk 2 above)
Comment (0xivanov) [reply hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/store/contract/HederaEvmStackedWorldStateUpdater.java]: Fixed.

File: hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/store/contract/HederaEvmStackedWorldStateUpdater.java (hunk 1 above)
Comment (0xivanov) [reply hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/store/contract/HederaEvmStackedWorldStateUpdater.java]: Fixed

File: hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/store/contract/HederaEvmStackedWorldStateUpdater.java (hunk 3 above)
Comment (0xivanov) [reply hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/store/contract/HederaEvmStackedWorldStateUpdater.java]: Fixed.

File: hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/store/contract/AbstractEvmStackedLedgerUpdater.java [hunk 4]
```diff
@@ -42,4 +47,37 @@ public UpdateTrackingAccount<A> getForMutation(Address address) {
...
+        // partially copied from services
+        final var wrapped = wrappedWorldView();
+        for (final var updatedAccount : getUpdatedAccounts().values()) {
+            var mutable = wrapped.getUpdatedAccounts().get(updatedAccount.getAddress());
+            if (mutable == null) {
+                mutable = updatedAccount.getWrappedAccount();
+                if (mutable == null) {
+                    mutable = new UpdateTrackingAccount<>(updatedAccount.getAddress(), null);
+                }
+                wrapped.getUpdatedAccounts().put(mutable.getAddress(), mutable);
```
Comment (steven-sheehy) [inline hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/store/contract/AbstractEvmStackedLedgerUpdater.java:68]: This seems to duplicate logic from https://github.com/hashgraph/hedera-mirror-node/pull/6066 but does not have that branch as its base. How do we plan to merge these two PRs?

**Review by steven-sheehy — DISMISSED** [review]
LGTM

File: hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/store/contract/AbstractEvmStackedLedgerUpdater.java (hunk 4 above)
Comment (IvanKavaldzhiev) [reply hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/store/contract/AbstractEvmStackedLedgerUpdater.java:68]: We should first merge #6066 and then take the changes into this one.

Comment (steven-sheehy) [reply hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/store/contract/AbstractEvmStackedLedgerUpdater.java:68]: Ok, that's merged. This one can be updated.

**Review by steven-sheehy — APPROVED** [review]
